### PR TITLE
#196 Button sizes.

### LINKED ISSUE #196: [Enhancement] Button sizes
**Is your enhancement request related to a problem? Please describe.**
Right now we only have one button (40 dp height) and one FAB in Kirby but we are going to have several different buttons in Kirby.

Here is a list:

40 dp height normal button (Primary and secondary) maybe with icons also
32dp height normal button (Primary, secondary and tertiary) with and without icons
40 dp circle button
32 dp circle button
56 dp FAB

**Describe the solution you'd like**
We would like those buttons ;) and also an overview in cookbook of course.

**Additional context**
Skærmbillede 2019-04-01 kl  14 43 03

### PR DESCRIPTION
**Pull request by jonasjensen-brolstark**
Closes #196

### CODE REVIEW

**Comment by jonasjensen-brolstark** [comment]
I have added a `size` directive, as i can see that some other kirby components also will need to have a `size` option.

File: src/app/showcase/button-showcase/button-showcase.component.ts [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
+      name: 'colortype',
+      description: 'Sets which color the button should use from the theme palette.',
+      defaultValue: 'primary',
+      inputValues: ['primary', 'secondary'],
+    },
+    {
+      name: 'size',
+      description: 'Sets the size of the button.',
+      defaultValue: 'medium',
+      inputValues: ['small', 'medium', 'large'],
```
Comment (jakobe) [inline src/app/showcase/button-showcase/button-showcase.component.ts]:
```suggestion
      inputValues: ['sm', 'md', 'lg'],
```

File: src/app/showcase/button-showcase/button-showcase.component.ts [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+    {
+      name: 'colortype',
+      description: 'Sets which color the button should use from the theme palette.',
+      defaultValue: 'primary',
+      inputValues: ['primary', 'secondary'],
+    },
+    {
+      name: 'size',
+      description: 'Sets the size of the button.',
+      defaultValue: 'medium',
```
Comment (jakobe) [inline src/app/showcase/button-showcase/button-showcase.component.ts]:
```suggestion
      defaultValue: 'md',
```

File: src/kirby/components/button/button.component.scss [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
+  }
+
+  &.large {
+    font-size: font-size('s');
+    height: size('xxl');
+  }
 
   &[expand='block'] {
-    display: block;
+    display: inline-block;
```
Comment (jakobe) [inline src/kirby/components/button/button.component.scss]: ?? But then it's not block anymore... :)

File: src/kirby/directives/size/size.directive.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  selector: 'Button[size], button[size]',
+})
+export class SizeDirective {
+  @HostBinding('class.small')
+  isSmallSize: boolean;
+  @HostBinding('class.medium')
+  isMediumSize: boolean;
+  @HostBinding('class.large')
+  isLargeSize: boolean;
+  @Input() set size(size: 'small' | 'medium' | 'large') {
```
Comment (jakobe) [inline src/kirby/directives/size/size.directive.ts]:
```suggestion
  @Input() set size(size: 'sm' | 'md' | 'lg') {
```

**Pull request merged**